kubetest2-aks/deployer/up.go

```diff
@@ -56,6 +59,70 @@ type UpOptions struct {
 	K8sVersion       string `flag:"k8sVersion" desc:"--k8sVersion flag for cluster Kubernetes version"`
 }
 
+type enableCustomFeaturesPolicy struct{}
+
+type changeLocationPolicy struct{}
+
+type addCustomConfigPolicy struct {
+	customConfig string
+}
+
+func (p enableCustomFeaturesPolicy) Do(req *policy.Request) (*http.Response, error) {
+	req.Raw().Header.Add("AKSHTTPCustomFeatures", "Microsoft.ContainerService/EnableCloudControllerManager")
+	return req.Next()
+}
+
+func (p changeLocationPolicy) Do(req *policy.Request) (*http.Response, error) {
+	if strings.Contains(req.Raw().URL.String(), "Microsoft.ContainerService/locations") {
+		q := req.Raw().URL.Query()
+		q.Set("api-version", "2017-08-31")
```

Comment:
When you use a different version, it will tell you: "The resource type 'locations/operations' could not be found in the namespace 'Microsoft.ContainerService' for api version '2022-03-01'. The supported api-versions are '2016-03-30,2017-08-31'."